File: URI/uri-1914/cpp-uri-1914/whose_turn_is_it.cpp
#include <iostream>
 
using namespace std;
 
int main() {
 
	int iterations;
	cin >> iterations;
	for(int i = 0; i < iterations; i++){
		string p1, p1Choice, p2, p2Choice;
		cin >> p1 >> p1Choice >> p2 >> p2Choice;
		int p1Numb, p2Numb;
		cin >> p1Numb >> p2Numb;
		if(p1Choice == "PAR"){
			if((p1Numb + p2Numb) % 2 == 0) 	cout << p1 << "\n";
			else				cout << p2 << "\n";
		}
		else{
			if((p1Numb + p2Numb) % 2 == 0)	cout << p2 << "\n";
			else				cout << p1 << "\n";
		}
	}
 
    return 0;
}

File: URI/uri-1914/python-uri-1914/whose_turn_is_it.py
n = int(input())
for i in range(n):
    p1, p1Choice, p2, p2Choice = input().split(" ")
    p1Numb, p2Numb = input().split(" ")
    p1Numb, p2Numb = int(p1Numb), int(p2Numb)
    if(p1Choice.lower() == "par"):
        if((p1Numb + p2Numb) % 2 == 0):  print(p1)
        else:                            print(p2)
    else:
        if((p1Numb + p2Numb) % 2 == 0):  print(p2)
        else:                            print(p1)
